model/task/task.go

```diff
@@ -1058,7 +1058,11 @@ func (t *Task) SetHasCedarResults(hasCedarResults, failedResults bool) error {
 	}
 
 	if !t.DisplayOnly && t.IsPartOfDisplay() {
-		return t.DisplayTask.SetHasCedarResults(hasCedarResults, failedResults)
+		displayTask, err := t.GetDisplayTask()
+		if err != nil {
+			return errors.Wrap(err, "getting display task")
+		}
+		return displayTask.SetHasCedarResults(hasCedarResults, failedResults)
```

Comment:
If you want this to destructively modify t, you'll want to still use t.DisplayTask here

Reply:
`GetDisplayTask` returns a pointer to the display task, so it will modify it either way.